# PART 2: TARGET ACHIEVEMENT ANALYSIS

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/Sales_target_DD2E9B96A0.csv")
df.head()

Month of Order Date   Category   Target
0              Apr-18  Furniture  10400.0
1              May-18  Furniture  10500.0
2              Jun-18  Furniture  10600.0
3              Jul-18  Furniture  10800.0
4              Aug-18  Furniture  10900.0

In [3]:
df["Month of Order Date"] = pd.to_datetime(df["Month of Order Date"],format="%b-%y")
df.head()

Month of Order Date   Category   Target
0          2018-04-01  Furniture  10400.0
1          2018-05-01  Furniture  10500.0
2          2018-06-01  Furniture  10600.0
3          2018-07-01  Furniture  10800.0
4          2018-08-01  Furniture  10900.0

In [4]:
furniture_df = df[df["Category"]=="Furniture"].copy()
furniture_df.count()

Month of Order Date    12
Category               12
Target                 12
dtype: int64

In [5]:
furniture_df.sort_values(by="Month of Order Date",inplace = True)

In [6]:
furniture_df["Percentage Change(%)"] = round(furniture_df["Target"].pct_change()*100,2)
furniture_df = furniture_df.fillna(0)
furniture_df

Month of Order Date   Category   Target  Percentage Change(%)
0           2018-04-01  Furniture  10400.0                  0.00
1           2018-05-01  Furniture  10500.0                  0.96
2           2018-06-01  Furniture  10600.0                  0.95
3           2018-07-01  Furniture  10800.0                  1.89
4           2018-08-01  Furniture  10900.0                  0.93
5           2018-09-01  Furniture  11000.0                  0.92
6           2018-10-01  Furniture  11100.0                  0.91
7           2018-11-01  Furniture  11300.0                  1.80
8           2018-12-01  Furniture  11400.0                  0.88
9           2019-01-01  Furniture  11500.0                  0.88
10          2019-02-01  Furniture  11600.0                  0.87
11          2019-03-01  Furniture  11800.0                  1.72

In [7]:
Avg_furniture = furniture_df['Percentage Change(%)'].sum() / furniture_df.count()
Avg_furniture['Percentage Change(%)']

np.float64(1.0591666666666668)

These are percentage changes in target sales for furniture category and has average of  `1.059%`

In [8]:
df_sorted = df.sort_values(["Category","Month of Order Date"])
df_sorted["Percentage Change"] = round(df_sorted.groupby("Category")["Target"].pct_change()*100,2)
df_sorted = df_sorted.fillna(0)
df_sorted

Month of Order Date     Category   Target  Percentage Change
12          2018-04-01     Clothing  12000.0               0.00
13          2018-05-01     Clothing  12000.0               0.00
14          2018-06-01     Clothing  12000.0               0.00
15          2018-07-01     Clothing  14000.0              16.67
16          2018-08-01     Clothing  14000.0               0.00
17          2018-09-01     Clothing  14000.0               0.00
18          2018-10-01     Clothing  16000.0              14.29
19          2018-11-01     Clothing  16000.0               0.00
20          2018-12-01     Clothing  16000.0               0.00
21          2019-01-01     Clothing  16000.0               0.00
22          2019-02-01     Clothing  16000.0               0.00
23          2019-03-01     Clothing  16000.0               0.00
24          2018-04-01  Electronics   9000.0               0.00
25          2018-05-01  Electronics   9000.0               0.00
26          2018-06-01  Electronics   9000.0               0.00
27          2018-07-01  Electronics   9000.0               0.00
28          2018-08-01  Electronics   9000.0               0.00
29          2018-09-01  Electronics   9000.0               0.00
30          2018-10-01  Electronics   9000.0               0.00
31          2018-11-01  Electronics   9000.0               0.00
32          2018-12-01  Electronics   9000.0               0.00
33          2019-01-01  Electronics  16000.0              77.78
34          2019-02-01  Electronics  16000.0               0.00
35          2019-03-01  Electronics  16000.0               0.00
0           2018-04-01    Furniture  10400.0               0.00
1           2018-05-01    Furniture  10500.0               0.96
2           2018-06-01    Furniture  10600.0               0.95
3           2018-07-01    Furniture  10800.0               1.89
4           2018-08-01    Furniture  10900.0               0.93
5           2018-09-01    Furniture  11000.0               0.92
6           2018-10-01    Furniture  11100.0               0.91
7           2018-11-01    Furniture  11300.0               1.80
8           2018-12-01    Furniture  11400.0               0.88
9           2019-01-01    Furniture  11500.0               0.88
10          2019-02-01    Furniture  11600.0               0.87
11          2019-03-01    Furniture  11800.0               1.72

In [9]:
df_sorted["Signficant Fluctuation"] = df_sorted["Percentage Change"].apply(lambda x:"Static" if x == 0 else("High Increase" if x >= 1.5 else("Low Increase" if x <= 0.9 else "Normal")))
fluctuation_months = df_sorted[df_sorted["Signficant Fluctuation"]!= "Normal"]
fluctuation_months

Month of Order Date     Category   Target  Percentage Change  \
12          2018-04-01     Clothing  12000.0               0.00   
13          2018-05-01     Clothing  12000.0               0.00   
14          2018-06-01     Clothing  12000.0               0.00   
15          2018-07-01     Clothing  14000.0              16.67   
16          2018-08-01     Clothing  14000.0               0.00   
17          2018-09-01     Clothing  14000.0               0.00   
18          2018-10-01     Clothing  16000.0              14.29   
19          2018-11-01     Clothing  16000.0               0.00   
20          2018-12-01     Clothing  16000.0               0.00   
21          2019-01-01     Clothing  16000.0               0.00   
22          2019-02-01     Clothing  16000.0               0.00   
23          2019-03-01     Clothing  16000.0               0.00   
24          2018-04-01  Electronics   9000.0               0.00   
25          2018-05-01  Electronics   9000.0               0.00   
26          2018-06-01  Electronics   9000.0               0.00   
27          2018-07-01  Electronics   9000.0               0.00   
28          2018-08-01  Electronics   9000.0               0.00   
29          2018-09-01  Electronics   9000.0               0.00   
30          2018-10-01  Electronics   9000.0               0.00   
31          2018-11-01  Electronics   9000.0               0.00   
32          2018-12-01  Electronics   9000.0               0.00   
33          2019-01-01  Electronics  16000.0              77.78   
34          2019-02-01  Electronics  16000.0               0.00   
35          2019-03-01  Electronics  16000.0               0.00   
0           2018-04-01    Furniture  10400.0               0.00   
3           2018-07-01    Furniture  10800.0               1.89   
7           2018-11-01    Furniture  11300.0               1.80   
8           2018-12-01    Furniture  11400.0               0.88   
9           2019-01-01    Furniture  11500.0               0.88   
10          2019-02-01    Furniture  11600.0               0.87   
11          2019-03-01    Furniture  11800.0               1.72   

   Signficant Fluctuation  
12                 Static  
13                 Static  
14                 Static  
15          High Increase  
16                 Static  
17                 Static  
18          High Increase  
19                 Static  
20                 Static  
21                 Static  
22                 Static  
23                 Static  
24                 Static  
25                 Static  
26                 Static  
27                 Static  
28                 Static  
29                 Static  
30                 Static  
31                 Static  
32                 Static  
33          High Increase  
34                 Static  
35                 Static  
0                  Static  
3           High Increase  
7           High Increase  
8            Low Increase  
9            Low Increase  
10           Low Increase  
11          High Increase

In [10]:
grouped_fluctuation = fluctuation_months.groupby("Signficant Fluctuation")
for fluctuation_type, group in grouped_fluctuation:
  print(f"\nFluctuation Type:{fluctuation_type}")
  print(group[["Month of Order Date","Category","Target","Percentage Change"]])


Fluctuation Type:High Increase
   Month of Order Date     Category   Target  Percentage Change
15          2018-07-01     Clothing  14000.0              16.67
18          2018-10-01     Clothing  16000.0              14.29
33          2019-01-01  Electronics  16000.0              77.78
3           2018-07-01    Furniture  10800.0               1.89
7           2018-11-01    Furniture  11300.0               1.80
11          2019-03-01    Furniture  11800.0               1.72

Fluctuation Type:Low Increase
   Month of Order Date   Category   Target  Percentage Change
8           2018-12-01  Furniture  11400.0               0.88
9           2019-01-01  Furniture  11500.0               0.88
10          2019-02-01  Furniture  11600.0               0.87

Fluctuation Type:Static
   Month of Order Date     Category   Target  Percentage Change
12          2018-04-01     Clothing  12000.0                0.0
13          2018-05-01     Clothing  12000.0                0.0
14          2018-06-01  

In [11]:
df_sorted["Formatted Month"] = df_sorted["Month of Order Date"].dt.strftime("%B %Y")
df_sorted = df_sorted.drop(columns="Month of Order Date")
df_sorted

Category   Target  Percentage Change Signficant Fluctuation  \
12     Clothing  12000.0               0.00                 Static   
13     Clothing  12000.0               0.00                 Static   
14     Clothing  12000.0               0.00                 Static   
15     Clothing  14000.0              16.67          High Increase   
16     Clothing  14000.0               0.00                 Static   
17     Clothing  14000.0               0.00                 Static   
18     Clothing  16000.0              14.29          High Increase   
19     Clothing  16000.0               0.00                 Static   
20     Clothing  16000.0               0.00                 Static   
21     Clothing  16000.0               0.00                 Static   
22     Clothing  16000.0               0.00                 Static   
23     Clothing  16000.0               0.00                 Static   
24  Electronics   9000.0               0.00                 Static   
25  Electronics   9000.0               0.00                 Static   
26  Electronics   9000.0               0.00                 Static   
27  Electronics   9000.0               0.00                 Static   
28  Electronics   9000.0               0.00                 Static   
29  Electronics   9000.0               0.00                 Static   
30  Electronics   9000.0               0.00                 Static   
31  Electronics   9000.0               0.00                 Static   
32  Electronics   9000.0               0.00                 Static   
33  Electronics  16000.0              77.78          High Increase   
34  Electronics  16000.0               0.00                 Static   
35  Electronics  16000.0               0.00                 Static   
0     Furniture  10400.0               0.00                 Static   
1     Furniture  10500.0               0.96                 Normal   
2     Furniture  10600.0               0.95                 Normal   
3     Furniture  10800.0               1.89          High Increase   
4     Furniture  10900.0               0.93                 Normal   
5     Furniture  11000.0               0.92                 Normal   
6     Furniture  11100.0               0.91                 Normal   
7     Furniture  11300.0               1.80          High Increase   
8     Furniture  11400.0               0.88           Low Increase   
9     Furniture  11500.0               0.88           Low Increase   
10    Furniture  11600.0               0.87           Low Increase   
11    Furniture  11800.0               1.72          High Increase   

   Formatted Month  
12      April 2018  
13        May 2018  
14       June 2018  
15       July 2018  
16     August 2018  
17  September 2018  
18    October 2018  
19   November 2018  
20   December 2018  
21    January 2019  
22   February 2019  
23      March 2019  
24      April 2018  
25        May 2018  
26       June 2018  
27       July 2018  
28     August 2018  
29  September 2018  
30    October 2018  
31   November 2018  
32   December 2018  
33    January 2019  
34   February 2019  
35      March 2019  
0       April 2018  
1         May 2018  
2        June 2018  
3        July 2018  
4      August 2018  
5   September 2018  
6     October 2018  
7    November 2018  
8    December 2018  
9     January 2019  
10   February 2019  
11      March 2019

In [12]:
Summary_df = df_sorted.groupby("Category").apply(
    lambda x: pd.Series({
        "High Increase":",".join(x.loc[x["Signficant Fluctuation"] == "High Increase","Formatted Month"]),
        "Low Increase":",".join(x.loc[x["Signficant Fluctuation"] == "Low Increase","Formatted Month"]),
        "Static":",".join(x.loc[x["Signficant Fluctuation"] == "Static","Formatted Month"]),
    })
    ).reset_index()
Summary_df

<ipython-input-12-3dbd9d1adcc7>:1: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  Summary_df = df_sorted.groupby("Category").apply(


Category                       High Increase  \
0     Clothing              July 2018,October 2018   
1  Electronics                        January 2019   
2    Furniture  July 2018,November 2018,March 2019   

                               Low Increase  \
0                                             
1                                             
2  December 2018,January 2019,February 2019   

                                              Static  
0  April 2018,May 2018,June 2018,August 2018,Sept...  
1  April 2018,May 2018,June 2018,July 2018,August...  
2                                         April 2018

The furniture category shows a single static value with target sales increasing gradually each month, which is good. Other categories have multiple static values, indicating a lack of predictive and historical data analysis.

* Targets were not adjusted based on previous performance or market trends. Analyzing historical performance and making predictions to set target sales will lead to achievable targets.

* A significant jump in the target does not yield a positive result. Gradually increasing the target will lead to realistic sales growth.

* Conducting regular performance reviews helps you set realistic and achievable targets.